### PR TITLE
Support for read only tokens

### PR DESCRIPTION
**Pull request by acierto**

### CODE REVIEW

**Comment by acierto** [comment]
@K-Phoen What do you think about this change?

File: .gitignore [hunk 1]
```diff
@@ -9,6 +9,9 @@ lerna-debug.log*
 # Diagnostic reports (https://nodejs.org/api/report.html)
 report.[0-9]*.[0-9]*.[0-9]*.[0-9]*.json
 
+# Intellij IDEA configuration's folder
+.idea
```
Comment (K-Phoen) [inline .gitignore:13]: There's no reason to include this in this file as a global .gitignore should be the one excluding IDE-specific things :)

File: config.d.ts [hunk 2]
```diff
@@ -13,6 +13,13 @@ export interface Config {
          */
         proxyPath?: string;
 
+        /**
+         * If OpsGenie token has only read rights, you have to set it to true.
+         * Some options would be disabled on a view part.
```
Comment (K-Phoen) [inline config.d.ts:18]: What about:

> Enables "read-only" mode on the plugin. In this mode, actions such as acknowledging  or closing an alert will be disabled.
> Enable this if you only want to expose Opsgenie-originated information or if the API token you use has only read permissions.

File: src/api.ts [hunk 3]
```diff
@@ -201,6 +210,10 @@ export class OpsgenieApi implements Opsgenie {
     return `${this.domain}/settings/users/${userId}/detail`;
   }
 
+  isReadOnly(): boolean {
+    return this.readOnly;
+  }
```
Comment (K-Phoen) [inline src/api.ts:215]: To make sure we enforce this setting, could we add a guard on all non-readonly methods in this class? 
`acknowledgeAlert()` and `closeAlert()` should throw an exception if they are somehow called and the readonly flag is true.

File: yarn.lock [hunk 4]
```diff
@@ -2,18 +2,12 @@
...
-"@adobe/css-tools@^4.0.1":
-  version "4.0.1"
-  resolved "https://registry.yarnpkg.com/@adobe/css-tools/-/css-tools-4.0.1.tgz#b38b444ad3aa5fedbb15f2f746dcd934226a12dd"
-  integrity sha512-+u76oB43nOHrF4DDWRLWDCtci7f3QJoEBigemIdIeTi1ODqjx6Tad9NCVnPRwewWlKkVab5PlK8DCtPTyX7S8g==
-
 "@ampproject/remapping@^2.1.0":
-  version "2.2.0"
-  resolved "https://registry.yarnpkg.com/@ampproject/remapping/-/remapping-2.2.0.tgz#56c133824780de3174aed5ab6834f3026790154d"
-  integrity sha512-qRmjj8nj9qmLTQXXmaR1cck3UXSRMPrbsLJAasZpF+t3riI71BXed5ebIOYwQntykeZuhjsdweEc9BxH5Jc26w==
+  version "2.1.2"
```
Comment (K-Phoen) [inline yarn.lock:6]: Is there any reason for this file to have changed? If not, could it be removed from the PR's diff?

**Review by K-Phoen — COMMENTED** [review]
I think this PR can be quite useful, thanks for submitting it!

I left a few small comments which once addressed will make it good to merge :)

File: .gitignore (hunk 1 above)
Comment (acierto) [reply .gitignore:13]: I can remove it sure.
Added here, as it's quite common to exclude `.idea` in `.gitignore` :)

File: yarn.lock (hunk 4 above)
Comment (acierto) [reply yarn.lock:6]: That what yarn generates ...

Ok, can revert it

**Comment by acierto** [comment]
@K-Phoen Addressed all comments

**Comment by K-Phoen** [comment]
Awesome, thanks :muscle:

**Pull request merged**